apps/dashboard/src/components/conditions-editor/variable-select.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { useMemo, useState, useRef } from 'react';
+import { CheckIcon } from '@radix-ui/react-icons';
+
+import { Code2 } from '@/components/icons/code-2';
+import { Popover, PopoverAnchor, PopoverContent } from '@/components/primitives/popover';
+import { InputPure, InputRoot, InputWrapper } from '@/components/primitives/input';
+import { AUTOCOMPLETE_PASSWORD_MANAGERS_OFF } from '@/utils/constants';
+import { cn } from '@/utils/ui';
+
+const KeyboardItem = ({ children, className }: { children: React.ReactNode; className?: string }) => {
+  return (
+    <span
+      className={cn(
+        'text-foreground-400 shadow-xs text-paragraph-2xs flex h-5 w-5 items-center justify-center rounded-[6px] border border-neutral-200 px-2 py-1 font-light',
+        className
+      )}
+    >
+      {children}
+    </span>
+  );
+};
+
+type VariablesListProps = {
+  options: Array<{ label: string; value: string }>;
+  onSelect: (value: string) => void;
+  selectedValue?: string;
+};
+
+const VariablesList = React.forwardRef<HTMLInputElement, VariablesListProps>(
+  ({ options, onSelect, selectedValue }, ref) => {
+    const [hoveredOptionIndex, setHoveredOptionIndex] = useState(-1);
+    const listRef = useRef<HTMLUListElement>(null);
+
+    const scrollToOption = (index: number) => {
+      if (!listRef.current) return;
+
+      const listElement = listRef.current;
+      const optionElement = listElement.children[index] as HTMLLIElement;
+
+      if (optionElement) {
+        const containerHeight = listElement.clientHeight;
+        const optionTop = optionElement.offsetTop;
+        const optionHeight = optionElement.clientHeight;
+
+        if (optionTop < listElement.scrollTop) {
+          // Scroll up if option is above visible area
+          listElement.scrollTop = optionTop;
+        } else if (optionTop + optionHeight > listElement.scrollTop + containerHeight) {
+          // Scroll down if option is below visible area
+          listElement.scrollTop = optionTop + optionHeight - containerHeight;
+        }
+      }
+    };
+
+    const next = () => {
+      if (hoveredOptionIndex === -1) {
+        setHoveredOptionIndex(0);
+        scrollToOption(0);
+      } else {
+        setHoveredOptionIndex((oldIndex) => {
+          const newIndex = oldIndex === options.length - 1 ? 0 : oldIndex + 1;
+          scrollToOption(newIndex);
+          return newIndex;
+        });
+      }
+    };
+
+    const prev = () => {
+      if (hoveredOptionIndex === -1) {
+        setHoveredOptionIndex(options.length - 1);
+        scrollToOption(options.length - 1);
+      } else {
+        setHoveredOptionIndex((oldIndex) => {
+          const newIndex = oldIndex === 0 ? options.length - 1 : oldIndex - 1;
+          scrollToOption(newIndex);
+          return newIndex;
+        });
+      }
+    };
+
+    return (
+      <div className="flex flex-col">
+        <input
```

Comment:
The hidden input manages the focus and keyboard navigation. I've tried with `cmdk,` but it doesn't work well and doesn't allow you to control the focus. Let me explain: when the input is focused, we want to show the dropdown but not focus on it so the input stays focused, then, when the keyboard navigation is used, we want to focus on the list and navigate through the list items.